integration/spark/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -0,0 +1,173 @@
+    var includedDestColumns = false
+    var includedDestRelation = false
+    var addedTupleId = false
+
+    def prepareTargetReleation(relation: UnresolvedRelation): Subquery = {
+      val tupleId = UnresolvedAlias(Alias(UnresolvedFunction("getTupleId",
+        Seq.empty, isDistinct = false), "tupleId")())
+      val projList = Seq(
+        UnresolvedAlias(UnresolvedStar(table.alias)), tupleId)
+      // include tuple id and rest of the required columns in subqury
+      Subquery(table.alias.getOrElse(""), Project(projList, relation))
+    }
+    // get the un-analyzed logical plan
+    val targetTable = prepareTargetReleation(table)
+    val selectPlan = org.apache.spark.sql.SQLParser.parse(selectStmt, sqlContext) transform {
+      case Project(projectList, child) if (!includedDestColumns) =>
+        includedDestColumns = true
+        if (projectList.size != columns.size) {
+          sys.error("Number of source and destination columns are not matching")
+        }
+        val renamedProjectList = projectList.zip(columns).map{ case(attr, col) =>
+          attr match {
+            case UnresolvedAlias(child) =>
+              UnresolvedAlias(Alias(child, col + "-updatedColumn")())
+            case _ => attr
+          }
+        }
+        val list = Seq(
+          UnresolvedAlias(UnresolvedStar(table.alias))) ++ renamedProjectList
+        Project(list, child)
+      case Filter(cond, child) if (!includedDestRelation) =>
+        includedDestRelation = true
+        Filter(cond, Join(child, targetTable, Inner, None))
+      case r @ UnresolvedRelation(t, a) if (!includedDestRelation &&
+        t != table.tableIdentifier) =>
+        includedDestRelation = true
+        Join(r, targetTable, Inner, None)
+    }
+    val updatedSelectPlan = if (!includedDestRelation) {
+      // special case to handle self join queries
+      // Eg. update tableName  SET (column1) = (column1+1)
+      selectPlan transform {
+        case relation: UnresolvedRelation if (table.tableIdentifier == relation.tableIdentifier &&
+          addedTupleId == false) =>
+          addedTupleId = true
+          targetTable
+      }
+    } else {
+      selectPlan
+    }
+    val finalPlan = if (filter.length > 0) {
+      val alias = table.alias.getOrElse("")
+      var transformed: Boolean = false
+      // Create a dummy projection to include filter conditions
+      SQLParser.parse("select * from  " +
+        table.tableIdentifier.mkString(".") + " " + alias + " " + filter, sqlContext)  transform {
+        case UnresolvedRelation(t, Some(a)) if (
+          !transformed && t == table.tableIdentifier && a == alias) =>
+          transformed = true
+          // Add the filter condition of update statement  on destination table
+          Subquery(alias, updatedSelectPlan)
+      }
+    } else {
+      updatedSelectPlan
+    }
+    val tid = CarbonTableIdentifierImplicit.toTableIdentifier(table.tableIdentifier)
```

Comment:
@ravikiran23 @jackylk UnresolvedRelation.tableIdentifier is a 'Seq[String]' type in Spark-1.5 while in Spark-1.6 it's a 'TableIdentifier' type, so compile unsuccessfully with Spark-1.6.